fluxc/src/main/java/org/wordpress/android/fluxc/logging/FluxCCrashLoggerProvider.kt

```
@@ -0,0 +1,19 @@
+package org.wordpress.android.fluxc.logging
+
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T
+
+object FluxCCrashLoggerProvider {
+    var crashLogger: FluxCCrashLogger? = null
```

Shell we make access to this prop thread-safe? 

Also, you use "private set" but `initLogger` is the same as public, any reason behind that?

---

> Shell we make access to this prop thread-safe?

That's what I meant by the naive implementation - I don't think it's worth making it thread safe. It should be called only once on application startup so it'll be called from a UI thread. The access doesn't matter as it's thread safe by nature. Wdyt?

> Also, you use "private set" but initLogger is the same as public, any reason behind that?

Yes, if we decide to do more stuff in the init, we won't need to change the clients. If it was a pure setter, doing more stuff in the init would felt weird. Wdyt?

---

> That's what I meant by the naive implementation - I don't think it's worth making it thread safe. It should be called only once on application startup so it'll be called from a UI thread. The access doesn't matter as it's thread safe by nature. Wdyt?

If you write from one thread, there is no guarantee that this change will be visible from another thread. So I would at least mark it `volatile`

And strictly speaking 

> It should be called only once on application startup

This is not ensured or declared anyhow in the current implementation. If you think that it has to be initiated only once, maybe throw an exception if someone tries to init it a second time?

---

> If you write from one thread, there is no guarantee that this change will be visible from another thread. So I would at least mark it volatile

I can't imagine a scenario in which the OS/CPU would decide to put this variable into a cache before it's used for the first time. All subsequent usages are not changing the value of the variable - and imo even then the OS/CPU won't put this particular variable into the cache. `Volatile` is imo an overkill but I don't mind adding it - this is not a performance hungry code anyway -> updated in b4de11e6d6e515df71336f294a0dfa41495de54d

> And strictly speaking
> > It should be called only once on application startup

Again, same as with the Thread safety - I personally think that adding this adds unnecessary complexity which won't bring us anything at all. If the clients want to invoke the method multiple times for some very weird reason it's their decision - the code is clear enough that it should be used just once imo, the method is called `init = initialize`. Having said that I don't mind adding it -  updated in d95c8b5cf59c2a796dee39b7c7ce0f6eb035d0ed.

P.S. Introducing a double locking mechanism feels too much so unless you really think it's worth it, I'd vote for keeping the code simple over absolutely safe.

---

This is a library which can be used by other clients, that's why I think it better to keep it on the safe side, because you never know how people will use this, so better to be safe than sorry =)

---

Updated in https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2421/commits/0b8e8594d94f9307c0480a56484ae85612ee0fdd.